examples/seismic/acoustic/wavesolver.py

```diff
@@ -102,15 +111,21 @@ def forward(self, src=None, rec=None, u=None, m=None, save=False, **kwargs):
                          space_order=self.space_order,
                          dtype=self.model.dtype)
 
+        kwargs.update({'u': u})
+
         # Pick m from model unless explicitly provided
-        if m is None:
-            m = m or self.model.m
+        if m is not None:
```

Comment:
Definitely, somehow, we should catch all this ``if m is not None: ...`` **before** this point, otherwise we end up with lots of duplicated code